ui/app/ducks/swaps/swaps.js

```diff
@@ -429,7 +429,11 @@ export const fetchQuotesAndSetQuoteState = (
     const contractExchangeRates = getTokenExchangeRates(state);
 
     let destinationTokenAddedForSwap = false;
-    if (toTokenSymbol !== 'ETH' && !contractExchangeRates[toTokenAddress]) {
+    if (
+      toTokenAddress &&
```

Comment:
This is necessary because the next two lines would allow `const selectedToToken =  {}` to add a token, which we definitely don't want.

Reply:
This looks like a fine change to protect against errors. However, we should never be in a situation where there is no `toTokenAddress`. If so, requesting quotes should fail.

In the issue description, it was described how editing the code so that `toTokenAddress` is undefined results in the described bug, which occurs after confirming the swap. So it sounds like requesting quotes succeeded (otherwise there wouldn't be a quote to submit on the view quote screen).

So that leads me to believe that there is some other value that is being used for the token address in the request for quote. Or `swapsTokens` contains a token that has an `undefined` address.

```
const destinationTokenInfo =
      swapsTokens?.find(({ address }) => address === toTokenAddress) ||
      selectedToToken;
 ```
 
 It would be good if we could verify exactly how the request for quotes succeeds when `const selectedToToken = {}` and what the address is in that case.

Reply:
We could remove the `{}` fallback from `selectedFromToken` and `selectedToToken` that existed from the initial swaps merge, and remove the `toTokenAddress` and `fromTokenAddress` checks in this PR.

If we should never get there without these properties, we should probably throw an error here, right?

Reply:
> If we should never get there without these properties, we should probably throw an error here, right?

Right, but what I want to know is how was it possible to request quotes without these properties? Requesting quotes when from or two are empty strings or undefined would return 0 quotes.

Reply:
I think this can be addressed in another PR, as the change to `actions.js` handles the objectives of this PR.